app/model/view/AssignmentViewFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace App\Model\View;
+
+use App\Model\Entity\Assignment;
+use App\Model\Entity\LocalizedExercise;
+use App\Security\ACL\IAssignmentPermissions;
+
+class AssignmentViewFactory {
+  /** @var IAssignmentPermissions */
+  private $assignmentAcl;
+
+  public function __construct(IAssignmentPermissions $assignmentAcl) {
+    $this->assignmentAcl = $assignmentAcl;
+  }
+
+  public function getAssignment(Assignment $assignment) {
+    return [
+      "id" => $assignment->getId(),
+      "version" => $assignment->getVersion(),
+      "isPublic" => $assignment->isPublic(),
+      "createdAt" => $assignment->getCreatedAt()->getTimestamp(),
+      "updatedAt" => $assignment->getUpdatedAt()->getTimestamp(),
+      "localizedTexts" => $assignment->getLocalizedTexts()->map(function (LocalizedExercise $text) use ($assignment) {
+        $data = $text->jsonSerialize();
+        if (!$this->assignmentAcl->canViewPrivateData($assignment)) {
```

Comment:
In my opinion `canViewPrivateData` (aka. private data) is really misleading here. As it is not private in the sense only visible to the author of assignment, but only visible to the supervisors. A better naming would be on point here.